hugegraph-llm/src/hugegraph_llm/utils/anchor.py

```
@@ -0,0 +1,28 @@
+
+def get_project_root() -> Path:
+    """Returns the Path object of the project root directory"""
+    current_path = Path(__file__).resolve()
+    for parent in current_path.parents:
+        if (parent / "pyproject.toml").exists() or (parent / ".git").exists():
+            return parent
+    # Raise an error if no project root is found
+    raise RuntimeError(
+        "Project root could not be determined. "
+        "Ensure that 'pyproject.toml' or '.git' exists in the project directory."
+    )
```

Pylint (3.3.7)

[convention] Final newline missing

(C0304)